src/transform.ts

```diff
@@ -102,9 +102,10 @@ export async function compileFile(
   const [clientScript, bindings] = clientScriptResult
   clientCode += clientScript
 
-  // script ssr only needs to be performed if using <script setup> where
-  // the render fn is inlined.
-  if (descriptor.scriptSetup) {
+  // script ssr needs to be performed if :
+  // 1.using <script setup> where the render fn is inlined.
+  // 2.using cssVars, as it do not need to be injected during SSR.
+  if (descriptor.scriptSetup || descriptor.cssVars.length > 0) {
```

Comment:
If cssVars is used,clientScript is not the same as ssrScript because we do not inject useCssVars into ssr.